src/core/analysis-context.ts

```diff
@@ -58,10 +61,15 @@ export class AnalysisContext {
   private _parsers = new Map<string, Parser<ParsedDocument<any, any>>>([
     ['html', new HtmlParser()],
     ['js', new JavaScriptParser()],
+    ['ts', new TypeScriptPreparser()],
     ['css', new CssParser()],
     ['json', new JsonParser()],
   ]);
 
+  private _languageAnalyzers = new Map<string, LanguageAnalyzer<any>>([
+    ['ts', new TypeScriptAnalyzer(this)],
```

Comment:
Oh definitely. The only driving reason for these additions is better JS analysis. We're using the TypeScript compiler for that because it seems to have the better API + inference capabilities than the alternatives. I'm only wiring it in for `ts` files for now so that we can get scanners ported and test coverage up before switching `js` to use them.